modules/apps/27-interchain-accounts/keeper/handshake.go

```diff
@@ -31,10 +32,25 @@ func (k Keeper) OnChanOpenInit(
 	version string,
 ) error {
 	if order != channeltypes.ORDERED {
-		return sdkerrors.Wrapf(channeltypes.ErrInvalidChannelOrdering, "invalid channel ordering: %s, expected %s", order.String(), channeltypes.ORDERED.String())
+		return sdkerrors.Wrapf(channeltypes.ErrInvalidChannelOrdering, "expected %s channel, got %s", channeltypes.ORDERED, order)
 	}
+
+	connSequence, err := types.ParseControllerConnSequence(portID)
+	if err != nil {
+		return sdkerrors.Wrapf(err, "expected format <app-version%[1]sconn-seq%[1]scounterparty-conn-seq%[1]sowner>, got %s", types.Delimiter, portID)
+	}
+
+	counterpartyConnSequence, err := types.ParseHostConnSequence(portID)
+	if err != nil {
+		return sdkerrors.Wrapf(err, "expected format <app-version%[1]sconn-seq%[1]scounterparty-conn-seq%[1]sowner>, got %s", types.Delimiter, portID)
+	}
+
+	if err := k.validateConnectionParams(ctx, channelID, portID, connSequence, counterpartyConnSequence); err != nil {
+		return sdkerrors.Wrapf(err, "failed to validate controller port (%s)", portID)
```

Comment:
maybe the error message should say connection params? Otherwise maybe we should rename the function to `validateControllerPort`?

Reply:
Sure thing! Any preference on this before I update?

Reply:
I think `validateControllerPort` makes the most sense to me. It will likely align closer with the ICS spec

Reply:
Cool, I've updated to `validateControllerPort`. My only gripe with this, and perhaps its overly pedantic, is that we never actually pass the controller port to this function. We pass the self port ID to retrieve the channel which is obviously going to be different in the `OnChainOpenInit` and `OnChanOpenTry` callbacks. 

We could... 🤔 

1. Call it `validateControllerPortParams` and leave the args the same.
2. Move the conn sequence parsing into the func and pass the controller port as an arg

The only problem I see with opt 2 is that having two port ID args may be confusing(?)

Either way I'm happy to leave it as is now if you are 😄

Reply:
I vote for `validateControllerPortParams and leave the args the same`